libsolidity/ast/AST.cpp

```diff
@@ -508,7 +508,7 @@ set<VariableDeclaration::Location> VariableDeclaration::allowedDataLocations() c
 
 	if (!hasReferenceOrMappingType() || isStateVariable() || isEventParameter())
 		return set<Location>{ Location::Unspecified };
-	else if (isStateVariable() && isConstant())
+	else if (isStateVariable() && (isConstant() || immutable()))
```

Comment:
Would this `else if` ever be taken for `isStateVariable() == true`? I think the one in line 509 catches all those cases and this one will never match?

Reply:
Right, it seems this can be removed.